dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/validation/validator/relationship/ConstraintValueTypeValidator.java

```diff
@@ -0,0 +1,73 @@
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.tracker.validation.validator.relationship;
+
+import static org.hisp.dhis.tracker.validation.validator.relationship.ValidationUtils.relationshipItemValueType;
+
+import org.hisp.dhis.relationship.RelationshipConstraint;
+import org.hisp.dhis.relationship.RelationshipType;
+import org.hisp.dhis.tracker.TrackerImportStrategy;
+import org.hisp.dhis.tracker.bundle.TrackerBundle;
+import org.hisp.dhis.tracker.domain.Relationship;
+import org.hisp.dhis.tracker.domain.RelationshipItem;
+import org.hisp.dhis.tracker.validation.Reporter;
+import org.hisp.dhis.tracker.validation.ValidationCode;
+import org.hisp.dhis.tracker.validation.Validator;
+
+public class ConstraintValueTypeValidator implements Validator<Relationship>
+{
+
+    @Override
+    public void validate( Reporter reporter, TrackerBundle bundle, Relationship relationship )
+    {
+
+        bundle.getPreheat().getAll( RelationshipType.class ).stream()
+            .filter( relationship.getRelationshipType()::isEqualTo ).findFirst().ifPresent( relationshipType -> {
+                validateValueTypeExists( reporter, relationship, "from", relationship.getFrom(),
+                    relationshipType.getFromConstraint() );
+                validateValueTypeExists( reporter, relationship, "to", relationship.getTo(),
+                    relationshipType.getToConstraint() );
+            } );
+    }
+
+    private void validateValueTypeExists( Reporter reporter, Relationship relationship, String relSide,
+        RelationshipItem item, RelationshipConstraint constraint )
+    {
+        if ( relationshipItemValueType( item ) == null )
```

Comment:
The way I understood we want the validators to be, is we want them to run all the validations they contain in the simplest way possible. 
Having that in mind, the thing about adding this logic to the `MandatoryFieldsValidator`, is that in case `from` or `to` are not set, we would have to skip this validation, otherwise we would get a NPE. That's why, having to skip the validation, pushes me into thinking that this needs to be in a separate validator.